pkg/apis/machine/validation/awsmachineclass_test.go

```diff
@@ -378,7 +371,7 @@ var _ = Describe("AWSMachineClass Validation", func() {
 					},
 				},
 				{
```

Comment:
I think tests should be added for the device name regex validation.

Reply:
Yes, naming conventions also depend on virtualization types, and are documented here: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html. 

As discussed, for the scope of this PR, I think that it's sufficient to validate that the device name matches the `/dev/[sd/xvd]*` regex, and that there is no duplication of device names. In case of device name incompatibility with the VM type, I think that it's reasonable to fail on an AWS SDK error. 

The AWS EBS CSI Driver uses the `/dev/xvd` prefix : https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/pkg/cloud/devicemanager/manager.go#L29, and generates a suffix taking into consideration existing device names: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/pkg/cloud/devicemanager/allocator.go#L52. I expect that the aws provider extensions for gardener would generate device names using a logic similar to the EBS CSI Driver.

Reply:
Sure, sounds good to me.